string segmentation_method 
---
bool success 
